x/incentive/types/genesis.go

```diff
@@ -269,10 +275,41 @@ func validateFPHistoricalRewards(entries []FinalityProviderHistoricalRewardsEntr
 			return err
 		}
 	}
+
+	// Validate that each FP with current rewards has all historical periods from 0 to currentPeriod-1
+	for fpAddr, currentPeriod := range fpCurrentPeriods {
+		if currentPeriod > 0 {
+			periods, exists := addressPeriodMap[fpAddr]
+			if !exists {
+				return fmt.Errorf("finality provider %s has current rewards with period %d but no historical rewards", fpAddr, currentPeriod)
+			}
+
+			// Check for all periods from 0 to currentPeriod-1
+			for period := uint64(0); period < currentPeriod; period++ {
```

Comment:
nice! just note for the future, when the prune of rewards is done, this check might need modification